master/buildbot/worker/ec2.py

```diff
@@ -71,7 +73,7 @@ def __init__(self, name, password, instance_type, ami=None,
                 "Use of default value of 'keypair_name' of EC2LatentWorker "
                 "constructor is deprecated. Please explicitly specify value")
             keypair_name = 'latent_buildbot_slave'
-        if security_name is None:
+        if security_name is None and not subnet_id:
             reportDeprecatedWorkerNameUsage(
```

Comment:
1. I'll leave DeprecatedWorkerNameUsage as is.
2. I don't think it makes sense to have a default value for security_name when it is an option only used outside of a VPC, something Amazon is deprecating and moving away from.
3. Yes, you are correct. `security_name` and `subnet_id` options are mutually exclusive. (as per this snippet)

```
if security_name and subnet_id:
            raise ValueError(
                'security_name (EC2 classic security groups) is not supported '
                'in a VPC.  Use security_group_ids instead.')
```

They indicate whether or not we will be using classic AWS EC2 or AWS VPC. Some more details are [here](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-vpc.html). The most relevant section of that, in my opinion:

> If you created your account after 2013-12-04, it supports EC2-VPC only

My preference is to leave the change as is, if you have any thoughts I'm open to hearing them!